In [10]:
import numpy as np
import pandas as pd
import statistics

In [51]:
names = ['Greg',
         'Marcia',
         'Peter',
         'Jan',
         'Bobby',
         'Cindy',
         'Jessica']

brady = pd.DataFrame(index=names)
brady['age'] = [14,12,11,10,8,7,1]

In [52]:
brady

age
Greg      14
Marcia    12
Peter     11
Jan       10
Bobby      8
Cindy      7
Jessica    1

In [53]:
avg = brady['age'].mean()

In [54]:
med = statistics.median(brady['age'])

In [55]:
mod = statistics.mode(brady['age'])

StatisticsError: no unique mode; found 7 equally common values

In [57]:
v = brady['age'].var()

In [58]:
dev = np.std(brady['age'], ddof=1)

In [59]:
er = dev/np.sqrt(len(brady['age']))

In [60]:
avg

9.0

In [61]:
med

10

In [62]:
mod

8

In [63]:
v

18.0

In [64]:
dev

4.242640687119285

In [65]:
er

1.6035674514745462

Thinkful Unit 3 lesson 1 Drill


1)

    Mean = 9.86 
    Median = 10.00 
    Mode = 8.00 
    Variance = 7.48 
    Standard deviation = 2.73 
    Standard error = 1.03

2) 
I would choose the mean for the central tendency due to the low range of numbers and the fact that there are no extreme numbers that would skew the mean in either direction.

For variance, I would select the standard deviation.  Nearly all the ages are within one standard deviation of the mean.

3)
	*Changed values
    
	*Mean = 10.00 
	Median = 10.00 
	Mode = 8.00 
	*Variance = 6.33 
	*Standard deviation = 2.52 
	*Standard error = 0.95

4) 
	Mean = 9.00
    Median = 10.00
    Mode = None
    Variance = 18.0
    Standard deviation = 4.24
    Standard error = 1.60
	
I would choose the median as the for the central tendency do the the outlier from the one year old.  For the I would also switch to the standard error for the variance estimation

5)
	16.25%
